### PR TITLE
[PSK-30, PSK-66] Changes to Recipe and addition of RecipeVote component

### PR DESCRIPTION
**Pull request by Andrius-Sukys**
Main changes:

[RECIPE]
> Updated Recipe to store Subfooddit and User names.
> Got rid of Description since it is no longer used in front-end.
> Added additional endpoints to query by user and subfooddit instead of polymorphic queries.
> Added AggregateVote field to store Users' votes.

[RECIPEVOTE]
> Added a component concerning Recipe Votes.
> Recipe Votes are synced with Recipe AggregatedVotes field.
> Added Authorization to identify voting user.

[NAMING]
> Minor changes to naming of controllers for consistency.
> Swapped VoteType values so they make more sense elsewhere.

### CODE REVIEW

File: receptai.api/Repositories/RecipeVoteRepository.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        return recipeVoteModels;
+    }
+
+    public async Task<RecipeVote> CreateAsync(RecipeVote recipeVoteModel)
+    {
+        await _context.RecipeVotes.AddAsync(recipeVoteModel);
+        await _context.SaveChangesAsync();
+
+        await _recipeRepository.RecalculateVotesAsync(
+            recipeVoteModel.RecipeId);
```
Comment (lynxcs) [inline receptai.api/Repositories/RecipeVoteRepository.cs:44]: Wouldn't it make more sense to add new votes, then recalculate, then save changes?

File: receptai.api/Repositories/RecipeVoteRepository.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        if (recipeVoteModel is null)
+        {
+            return null;
+        }
+
+        _context.RecipeVotes.Remove(recipeVoteModel);
+        await _context.SaveChangesAsync();
+
+        await _recipeRepository.RecalculateVotesAsync(recipeId);
+
```
Comment (lynxcs) [inline receptai.api/Repositories/RecipeVoteRepository.cs:64]: Same thing here

File: receptai.api/Repositories/RecipeVoteRepository.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        if (existingRecipeVote is null)
+        {
+            return null;
+        }
+
+        existingRecipeVote.VoteType = recipeVoteDto.VoteType;
+        existingRecipeVote.VoteDate = DateTime.Now;
+
+        await _context.SaveChangesAsync();
+        await _recipeRepository.RecalculateVotesAsync(recipeId);
```
Comment (lynxcs) [inline receptai.api/Repositories/RecipeVoteRepository.cs:92]: and here

File: receptai.api/Controllers/RecipeVoteController.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+        var recipeVotes = await _recipeVoteRepository.GetByRecipeId(recipeId);
+
+        return Ok(recipeVotes.Select(rv => rv.ToRecipeVoteDto()).ToList());
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> Create(
+        [FromBody] CreateRecipeVoteRequestDto recipeVoteDto)
+    {
```
Comment (lynxcs) [inline receptai.api/Controllers/RecipeVoteController.cs]: should use [Authorize], user id taken from claims principal via User.GetId();

Comment (lynxcs) [reply receptai.api/Controllers/RecipeVoteController.cs]: same goes for other endpoints which take in user id and relate to doing actions "for" the user

File: receptai.api/Repositories/RecipeVoteRepository.cs (hunk 1 above)
Comment (Andrius-Sukys) [reply receptai.api/Repositories/RecipeVoteRepository.cs:44]: The change in list of votes must be saved first before getting it to recalculate votes, I think my solution is viable. Swapping them around does not work.

**Review by R0PK3 — APPROVED** [review]
Looks very clean now, gj

**Pull request merged**